src/lib/labs/scheduler.ts

```
@@ -0,0 +1,54 @@
+import { LabEquipment, EquipmentReservation, UserSafetyCertification } from '@/types/labEquipment';
+
+/**
+ * Validates whether a student is eligible to reserve a piece of lab / HPC equipment.
+ * Checks for safety certification prerequisites and time collisions.
+ */
+export function validateReservationEligibility(
+  equipment: LabEquipment,
+  studentCertifications: UserSafetyCertification[],
+  existingReservations: EquipmentReservation[],
+  proposedStart: Date,
+  proposedEnd: Date
```

ESLint

[error] Insert `,`

(prettier/prettier)